factcast-server-ui/src/main/java/org/factcast/server/ui/report/ReportQueryPage.java

```diff
@@ -0,0 +1,304 @@
+      final var heading = new H4("Select Starting Serial");
+      final var overlayContent = new VerticalLayout(heading, since, latestSerial, fromScratch);
+      overlayContent.setSpacing(false);
+      overlayContent.getThemeList().add("spacing-xs");
+      overlayContent.setAlignItems(Alignment.STRETCH);
+      serialHelperOverlay.add(overlayContent);
+
+      from.setWidth("auto");
+
+      fileNameField.setLabel("Report File Name");
+      fileNameField.setPlaceholder("events");
+      fileNameField.setMinLength(1);
+      fileNameField.setTooltipText("The name of the file generated by the report.");
+      fileNameField.setClearButtonVisible(true);
+      fileNameField.setSuffixComponent(new Span(".json"));
+      fileNameField.setValueChangeMode(ValueChangeMode.EAGER);
+      fileNameField.addValueChangeListener(
+          e -> {
+            if (!fileNameField.isEmpty()) {
+              queryBtn.setEnabled(true);
+              fileName = sanitizeFileName(e.getValue()) + ".json";
+            } else {
+              queryBtn.setEnabled(false);
+            }
+          });
+
+      add(from, fileNameField);
+    }
+
+    private String sanitizeFileName(String fileName) {
+      return fileName.replaceAll("[^a-zA-Z0-9-_]", "");
+    }
+  }
+
+  @NonNull
+  private HorizontalLayout queryButtons() {
+    queryBtn.addClickShortcut(Key.ENTER);
+    queryBtn.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
+    queryBtn.setDisableOnClick(true);
+    queryBtn.addClickListener(event -> runQuery());
+
+    final var resetBtn = new Button("Reset");
+    resetBtn.addClickListener(
+        event -> {
+          formBean.reset();
+          binder.readBean(formBean);
+          factCriteriaViews.rebuild();
+          queryBtn.setEnabled(true);
+        });
+
+    final var hl = new HorizontalLayout(queryBtn, resetBtn);
+    hl.setWidthFull();
+    hl.addClassName("label-padding");
+
+    return hl;
+  }
+
+  private void runQuery() {
+    try {
+      queryBtn.setEnabled(false);
+      binder.writeBean(formBean);
+      log.info("{} runs query for {}", userName, formBean);
+      // For now this will block the UI in case of long-running queries. Will be refactored in the
+      // future once the FactRepository is adapted.
+      List<Fact> dataFromStore = repo.fetchAll(formBean);
+      log.info("Found {} entries", dataFromStore.size());
+      if (!dataFromStore.isEmpty()) {
+        List<ObjectNode> processedFacts =
+            dataFromStore.stream().map(e -> jsonViewPluginService.process(e).fact()).toList();
+        trySave(processedFacts);
+      } else {
+        displayWarning(
+            "No data was found for this query and therefore report creation is skipped.");
+      }
+    } catch (ValidationException e) {
+      Notifications.warn(e.getMessage());
+    } catch (Exception e) {
+      Notifications.error(e.getMessage());
+    }
+    // Not re-enabling the queryReportBtn to not incentivise users to generate it
+    // multiple times.
```

Comment:
should we consider re-enabling it at least when the report is empty or, even better, when the filter is updated somehow? Otherwise one would need to reset the filter each time, or know that changing the report name actually re-enables the button (which seems more like a workaround)